### DISCUSSION TITLE
Browser MHA behavior

### DISCUSSION THREAD

**Discussion by a2geekgirl** [microsoft/MHA#318] [Q&A] [answered]
I'm working with a client who is a DoD contractor and your fantastic MHA website came up. The question came up as to where the data in the browser is processed and if this  tool is simply on the fly converting the inputted headers or if they are being cached anywhere?  I can see that the Outlook add in has some rights in order to pull headers, but my client only plans to use the  browser version. Can you confirm if this data is cached outside of the browser? If it is, that might limit what headers the client would run through the tools due to the sensitive nature of their business.

**Comment by stephenegriffin** [comment, marked as answer]
Everything is processed on the fly client side. Nothing is cached outside the browser.
